src/core/control.cc

```diff
@@ -191,7 +191,7 @@ static std::shared_ptr<AbstractNode> builtin_assign(const ModuleInstantiation *i
       LOG(message_group::Warning, inst->location(), context->documentRoot(), "Assignment without variable name %1$s", argument->toEchoStringNoThrow());
     } else {
       if (assignContext->lookup_local_variable(*argument.name)) {
-        LOG(message_group::Warning, inst->location(), context->documentRoot(), "Duplicate variable assignment %1$s = %2$s", *argument.name, argument->toEchoStringNoThrow());
+        LOG(message_group::Warning, inst->location(), context->documentRoot(), "Duplicate variable assignment %1$s = %2$s", quoteVar(*argument.name), argument->toEchoStringNoThrow());
```

Comment:
Not sure we want to quote just the variable here; it's the expression that's interesting.

Reply:
I debated that one, and would accept either way, but was persuaded by the original-issue case where the variable name is "type", which could in theory yield something like:
```
Duplicate variable assignment type = "number"
```

Reply:
or did you mean something that would put quotes around the entire assignment, like
```
Duplicate variable assignment "name = value"
```
?

Reply:
(Which would need a `quoteExpr()` function.)

Reply:
Yeah, I meant that we should quote the expression. But as you point put, the value could also be quoted, which could be confusing when using double quotes, so perhaps that's indeed a use-case for quoteExpression :) ..or leave unquoted for now and revisit later.

Reply:
I'd rather not dive into `quoteExpr()` today.  We could leave the variable name quoted as I have it, or revert the change for that particular case.  (I think there might have been two of them.)  Your call.

Reply:
BTW, I think the answer for `quoteExpr()` might be to use something like `>>>...<<<` around the expression, like
```
Duplicate variable assignment >>> name = ... some expression ... <<<
```

Reply:
I'm fine with leaving it,  but perhaps add a TODO comment there so we don't lose track